### PR TITLE
docs(gotchas): adds dhis2-release-lifecycles doc

### PR DESCRIPTION
**Pull request by paschalidi**
Hey all! 👋 

This is the doc I have put together after @amcgee schooled us on how we do releases. Also Phil helped a lot with his feedback knowledge on the topic. You can fine the whole thread on slack [here](https://dhis2.slack.com/archives/C0BP0RABF/p1590137822140900) 

Any feedback or thoughts are always welcomed! :) Also it looks to me that `gotchas` was the best place to add this doc, it might be thought that it is really not, so please let me know about it as well :)

### CODE REVIEW

File: gotchas/dhis2-release-lifecycles.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# DHIS2 release lifecycles
+
+This is a walk through on the DHIS2  software releases and its lifecycles. Starting off we are gonna communicate some basic concepts in order to build a common language that we will use to better explain the release cycle.
```
Comment (Philip-Larsen-Donnelly) [inline gotchas/dhis2-release-lifecycles.md:3]: "gonna" is slang - please use "going to"  :)

File: gotchas/dhis2-release-lifecycles.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+#### Frontend core applications
+
+DHIS2 implements and maintains different core web applications.  The core web apps also follow a branching convention, however that is slightly different from the one followed in the backend. Same as in the backend the `master` branch includes the latest of the implemented software and is always the next candidate for a major release. However for each released version of the software there is a branch that follows the `v{x}` pattern. Eg. `v35`.  
+
+_Note: There are some core frontend apps that are currently now following this convention. While writing this the exceptions are the [data-visualiser-app](https://github.com/dhis2/data-visualizer-app/) which is using a different convention and [a list of apps ](https://github.com/dhis2/dhis2-core/tree/master/dhis-2/dhis-web/dhis-web-apps) where feature toggling is being used._
```
Comment (Philip-Larsen-Donnelly) [inline gotchas/dhis2-release-lifecycles.md]: typo - should be "... currently **not** following ..."

File: gotchas/dhis2-release-lifecycles.md (hunk 1 above)
Comment (paschalidi) [reply gotchas/dhis2-release-lifecycles.md]: changed it :) thanks

File: gotchas/dhis2-release-lifecycles.md (hunk 2 above)
Comment (paschalidi) [reply gotchas/dhis2-release-lifecycles.md]: good catch Phil. thanks

**Comment by varl** [comment]
Perhaps this could be something to put on the blog instead?

**Comment by paschalidi** [comment]
yeah there is a discussion going on here @varl https://dhis2.slack.com/archives/C0BP0RABF/p1590137822140900.

**Review by amcgee — APPROVED** [review]
Nice work @paschalidi !!  Thanks for putting this together.  I made some minor edits for language and clarity, will merge now 🎉 .  I will also include a link to this from developers.dhis2.org!

**Pull request merged**